src/systems/ec/implementations/vanilla/packages/ec-component-inpage-navigation/ec-component-inpage-navigation.scss

```diff
@@ -0,0 +1,159 @@
+/**
+ * ECL Navigation inpage
+ * @define inpage-navigation
+ */
+
+// Import base
+@import '@ecl/ec-base/ec-base';
+
+@mixin ecl-inpage-navigation($inpage-link-border: $ecl-spacing-2-xs) {
+  .ecl-inpage-navigation {
+    margin: 0;
+    position: relative;
+    z-index: 1;
+  }
+
+  .ecl-inpage-navigation__trigger {
+    background: linear-gradient(
+      to left,
+      $ecl-color-blue-120 50px,
+      $ecl-color-primary 50px
+    );
+    border: 0;
+    color: $ecl-color-white-100;
+    display: none;
+    padding: $ecl-spacing-s $ecl-spacing-m;
+    position: relative;
+    text-align: left;
+    width: 100%;
+
+    &:focus {
+      outline: none;
+    }
+
+    .ecl-inpage-navigation--visible & {
+      display: block;
+    }
+
+    .ecl-inpage-navigation__trigger-current {
+      font: $ecl-font-prolonged-m;
+      font-weight: $ecl-font-weight-bold;
+      margin-right: $ecl-font-size-xl;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      white-space: nowrap;
+    }
+
+    .ecl-inpage-navigation__trigger-icon {
+      position: absolute;
+      right: 1em;
+    }
+
+    &[aria-expanded='true'] .ecl-inpage-navigation__trigger-icon {
+      transform: rotate(0deg);
+    }
+  }
+
+  .ecl-inpage-navigation__title {
+    color: $ecl-color-grey-100;
+    display: none;
+    font: $ecl-font-m;
+    padding: $ecl-spacing-xs calc(#{$ecl-spacing-s} + #{$inpage-link-border});
+    text-transform: uppercase;
+  }
+
+  .ecl-inpage-navigation__list {
+    border-top: 1px solid #4073af;
+    list-style-type: none;
+    margin: 0;
+    padding: $ecl-spacing-xl $ecl-spacing-m 0;
+
+    &[aria-hidden='true'] {
+      display: none;
+    }
+  }
+
+  .ecl-inpage-navigation__link {
```

Comment:
since you add this class to a `<Link>` component, you must make sure the styles from Link have already been imported. It's also true for the Icon component. It means:

1. add `@ecl/ec-component-link` and `@ecl/ec-component-icon` to the dependencies
2. add a check `@include check-imports(('ec-component-link', 'ec-component-icon'));` at the beginning of the file